bridges/snowbridge/pallets/system-frontend/src/lib.rs

```diff
@@ -105,49 +119,64 @@ pub mod pallet {
 		/// Convert versioned location failure
 		UnsupportedLocationVersion,
 		/// Check location failure, should start from the dispatch origin as owner
-		OwnerCheck,
+		InvalidAssetOwner,
 		/// Send xcm message failure
 		Send,
 		/// Withdraw fee asset failure
 		FundsUnavailable,
+		/// Convert to reanchored location failure
+		LocationConversionFailed,
 	}
 
 	#[pallet::call]
 	impl<T: Config> Pallet<T> {
 		/// Call `create_agent` to instantiate a new agent contract representing `origin`.
-		/// - `origin`: Can be any sovereign `Location`
 		/// - `fee`: Fee in Ether paying for the execution cost on Ethreum
 		#[pallet::call_index(1)]
 		#[pallet::weight(T::WeightInfo::create_agent())]
 		pub fn create_agent(origin: OriginFor<T>, fee: u128) -> DispatchResult {
```

Comment:
See my comment above, we shouldn't be hardcoding the bridgehub fee.

```suggestion
		/// - `relayer_fee`: Relayer fee in Ether for relaying message to Ethereum and dispatching it.
		/// - `acceptance_fee`: Fee in native relay token for accepting the order for delivery
		#[pallet::call_index(1)]
		#[pallet::weight(T::WeightInfo::create_agent())]
		pub fn create_agent(origin: OriginFor<T>, relayer_fee: u128, acceptance_fee: u128) -> DispatchResult {

```